CorsixTH/Src/th.h

```diff
@@ -26,59 +26,63 @@ SOFTWARE.
 #include <vector>
 
 //! Generic linked list class (for inheriting from)
-struct THLinkList
+class link_list
 {
-    THLinkList() ;
-    ~THLinkList();
+public:
+    link_list() ;
+    ~link_list();
```

Comment:
Given its purpose is inheriting (as stated in the comment), destructor should be virtual